plugins/inputs/proxmox/proxmox.go

```diff
@@ -0,0 +1,259 @@
+package proxmox
+
+import (
+	"encoding/json"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+var sampleConfig = `
+  ## API connection configuration. The API token was introduced in Proxmox v6.2. Required permissions for user and token: PVEAuditor role on /.
+  base_url = "https://localhost:8006/api2/json"
+  api_token = "USER@REALM!TOKENID=UUID"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  insecure_skip_verify = false
+
+  # HTTP response timeout (default: 5s)
+  response_timeout = "5s"
+`
+
+func (px *Proxmox) SampleConfig() string {
+	return sampleConfig
+}
+
+func (px *Proxmox) Description() string {
+	return "Provides metrics from Proxmox nodes (Proxmox Virtual Environment > 6.2)."
+}
+
+func (px *Proxmox) Gather(acc telegraf.Accumulator) error {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+	px.hostname = hostname
```

Comment:
I'm thinking we should do this once on Init. if you define a function with the signature `Init() error` it will be called once when the plugin is initialized.